client/components/course/Sidebar/Relationship.vue

```diff
@@ -63,19 +69,27 @@ export default {
     associations() {
       const ids = this.getAssociationIds(this.activity);
       return filter(this.options, it => includes(ids, it.id));
+    },
+    displayType() {
+      return this.allowedTypes.length > 1;
     }
   },
   methods: {
     ...mapActions(['update'], 'activities'),
     getCustomLabel(activity) {
-      return get(activity, 'data.name', '');
+      const name = get(activity, 'data.name', '');
+      const type = get(getLevel(activity.type), 'label');
+      return !this.displayType ? name : `${name} (${type})`;
     },
     getAssociationIds(activity) {
       return get(activity, `refs.${this.type}`, []);
     },
     onRelationshipChanged(associations) {
+      const associationIds = associations.id
+        ? [associations.id]
+        : map(associations, 'id');
```

Comment:
I think this would be more robust:
```js
const associationIds = Array.isArray(associations)
	? map(associations, 'id')
	: get(associations, 'id');
```